File: src/gestorAplicacion/Cuenta.java
package gestorAplicacion;

import java.util.ArrayList;

public abstract class Cuenta {

    protected static int numero = 10000000;
    protected int saldoTotal;
    protected int SaldoDisponible;
    protected Cliente titular;
    protected Prestamo prestamo;
    protected ArrayList<Bolsillo> misBolsillos = new ArrayList<>();
    protected int deuda;
    protected boolean multa;
    protected boolean estado;
    
    public Cuenta(Cliente titular) {
        this.titular = titular;
        this.estado = true;
        Cuenta.numero++;

    }


    public abstract void aumentarSaldo(int cantidad);

    public abstract void disminuirSaldo(int cantidad);

    public abstract void  solicitarPrestamo(int valorPrestamo, String tipoPrestamo);

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        Cuenta.numero = numero;
    }

    public int getSaldoTotal() {
        return saldoTotal;
    }

    public void setSaldoTotal(int saldoTotal) {
        this.saldoTotal = saldoTotal;
    }

    public int getSaldoDisponible() {
        return SaldoDisponible;
    }

    public void setSaldoDisponible(int saldoDisponible) {
        SaldoDisponible = saldoDisponible;
    }

    public Cliente getTitular() {
        return titular;
    }

    public void setTitular(Cliente titular) {
        this.titular = titular;
    }

    public Prestamo getPrestamo() {
        return prestamo;
    }

    public void setPrestamo(Prestamo prestamo) {
        this.prestamo = prestamo;
    }

    public int getDeuda() {
        return deuda;
    }

    public void setDeuda(int deuda) {
        this.deuda = deuda;
    }

    public boolean isMulta() {
        return multa;
    }

    public void setMulta(boolean multa) {
        this.multa = multa;
    }

    public boolean isEstado() {
        return estado;
    }

    public void setEstado(boolean estado) {
        this.estado = estado;
    }


    public ArrayList<Bolsillo> getAhorro() {
        return misBolsillos;
    }

    public void setAhorro(ArrayList<Bolsillo> ahorro) {
        this.misBolsillos = ahorro;
    }

}

File: Python/gestorAplicacion/cuenta.py
import random
class Cuenta:
    id = 1000

    def __int__(self, titular, saldo, tipoCuenta):
        self.titular = titular
        self.saldo = saldo
        self.tipoCuenta = tipoCuenta
        self.saldoTotal = 0
        self.saldoDisponible = 0
        self.misBolsillos = []
        self.estado = True
        self.numero = random.randint(1000,10000)
        Cuenta.id += 1

        def getTitular(self):
            return self.titular

        def setTitular(self, titular):
            self.titular = titular

        def getSaldo(self):
            return self.saldo

        def setSaldo(self, saldo):
            self.saldo = saldo

        def getTipoCuenta(self):
            return self.tipoCuenta

        def setTipoCuenta(self, tipoCuenta):
            self.tipoCuenta = tipoCuenta

        def getSaldoTotal(self):
            return self.saldoTotal

        def setSaldoTotal(self, saldoTotal):
            self.saldoTotal = saldoTotal

        def getSaldoDisponibles(self):
            return self.saldoDisponible

        def setSaldoDisponible(self, saldoDisponible):
            self.saldoDisponible= saldoDisponible

        def getMisBolsillos(self):
            return self.misBolsillos

        def setMisBolsillos(self, misBolsillos):
            self.misBolsillos = misBolsillos






